src/compiler/checker.ts

```diff
@@ -55,6 +55,7 @@ namespace ts {
         let cancellationToken: CancellationToken | undefined;
         let requestedExternalEmitHelpers: ExternalEmitHelpers;
         let externalHelpersModule: Symbol;
+        let isInTypeOf = false;

```

Comment:
Do you really need this? You can get this in `resolveNameHelper` by doing `findAncestor` on `location`?

Reply:
Yes, but climbing ancestors has been bad for performance in the past (#17831, #17721). 99% of the time the expression won't be in a `typeof` and we'll have to climb all the way to the root.

Reply:
There is already `isInTypeQuery`